lib/ansible/modules/windows/win_disk_management.ps1

```diff
@@ -0,0 +1,716 @@
+        }
+        else{
+                $Volume | Format-Volume @ParaVol -UseLargeFRS -ShortFileNameSupport $FileSystemShortNames -Force -Confirm:$false
+        }
+
+}
+elseif($FileSystem -eq "refs"){
+$Volume | Format-Volume @ParaVol -SetIntegrityStreams $FileSystemIntegrityStreams -Force -Confirm:$false
+}
+
+}
+
+$params = Parse-Args $args;
+
+# Create a new result object
+$result = [pscustomobject]@{changed=$false;log=[pscustomobject]@{};general_log=[pscustomobject]@{};parameters=[pscustomobject]@{};switches=[pscustomobject]@{}}
+
+## Extract each attributes into a variable
+# Find attributes
+$Size = Get-Attr -obj $params -name "disk_size" -resultobj $result -failifempty $true
+$FindPartitionStyle = Get-Attr -obj $params -name "partition_style_select" -default "raw" -ValidateSet "raw","mbr","gpt" -resultobj $result -failifempty $false
+$OperationalStatus = Get-Attr -obj $params -name "operational_status" -default "offline" -ValidateSet "offline","online" -resultobj $result -failifempty $false
+
+# Set attributes partition
+$SetPartitionStyle = Get-Attr -obj $params -name "partition_style_set" -default "gpt" -ValidateSet "gpt","mbr" -resultobj $result -failifempty $false
+$DriveLetter = Get-Attr -obj $params -name "drive_letter" -default "e" -resultobj $result -failifempty $false
+# Set attributes partition
+$FileSystem = Get-Attr -obj $params -name "file_system" -default "ntfs" -ValidateSet "ntfs","refs" -resultobj $result -failifempty $false
+$Label = Get-Attr -obj $params -name "label" -default "ansible_disk" -resultobj $result -failifempty $false
+$AllocUnitSize = Get-Attr -obj $params -name "allocation_unit_size" -default "4" -ValidateSet "4","8","16","32","64" -resultobj $result -failifempty $false
+$LargeFRS = Get-Attr -obj $params -name "large_frs" -default $false -type "bool" -resultobj $result -failifempty $false
+$ShortNames = Get-Attr -obj $params -name "short_names" -default $false -type "bool" -resultobj $result -failifempty $false
+$IntegrityStreams = Get-Attr -obj $params -name "integrity_streams" -default $false -type "bool" -resultobj $result -failifempty $false
+
+# Convert variable for disk size
+[int32]$DSize = [convert]::ToInt32($Size, 10)
+
+# Define script environment variables
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version 2
+
+# Rescan disks
+try{
+Invoke-Expression '"rescan" | diskpart' | Out-Null
+   }
+catch{
+Set-Attr $result.general_log "rescan_disks" "failed"
+}
+
+Set-Attr $result.general_log "rescan_disks" "successful"
+
+[hashtable]$ParamsDisk = @{
+                          DiskSize = $DSize
+                          PartitionStyle = $FindPartitionStyle
+                          OperationalStatus = $OperationalStatus
+                          }
+
+# Search disk
+try{
+$disk = Search-Disk @ParamsDisk
+}
+catch{
+Set-Attr $result.general_log "search_disk" "failed"
+Fail-Json $result $_
```

Comment:
Would be good to put some context around where this failed like `Fail-Json -obj $result -message "failed to get disk info for ...: $($_.Exception.Message)"`